packages/sdk/src/index.ts

```diff
@@ -279,13 +283,24 @@ export default class SDK {
 
   /**
    * Creates a Request instance that can be sent through the RPCh network
-   * @param origin
    * @param provider
    * @param body
    * @returns Request
    */
   public createRequest(provider: string, body: string): Request {
     if (!this.isReady) throw Error("SDK not ready to create requests");
+    let entryNodeScore: number = this.reliabilityScore.getScore(
+      this.entryNode!.peerId
+    );
+    const exclusionList: string[] = [];
+    while (entryNodeScore < 0.7 && entryNodeScore !== 0.2) {
+      exclusionList.push(this.entryNode!.peerId);
+      this.selectEntryNode(
```

Comment:
shouldn't this be awaited?

Reply:
also I would add some logs, this part seems important

Reply:
Logs added and await for selectEntryNode added as well